Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -60,6 +60,12 @@ + (instancetype)sharedInstance
 
 + (void)reset
 {
+#if !TARGET_OS_WATCH
```

Comment:
I think this could solve some flaky reachability tests. IMO, this should be in an extra PR, so we can merge it right away. I don't want to steal your credit, so I leave it up to you if you want to open a PR or not. Otherwise, I can do it.

Reply:
Yep, that's exactly what it was for. In general I did plan to split up some of the fixes in here... I've just been trying to plow through anything that could possibly be causing problems in the TSAN tests for now. Easier to test the combinations together, then split them apart for discussion/historical purposes afterwards!

Reply:
See https://github.com/getsentry/sentry-cocoa/pull/3356